tests/components/test_prometheus.py

```diff
@@ -0,0 +1,59 @@
+"""The tests for the Prometheus exporter."""
+import unittest
+
+import requests
+
+from homeassistant import bootstrap, const
+import homeassistant.core as ha
+import homeassistant.components.prometheus as prometheus
+import homeassistant.components.http as http
+
+from tests.common import get_test_instance_port, get_test_home_assistant
+
+API_PASSWORD = "test1234"
+SERVER_PORT = get_test_instance_port()
+PROMETHEUS_URL = "http://127.0.0.1:{}/api/prometheus".format(SERVER_PORT)
+HA_HEADERS = {
+    const.HTTP_HEADER_HA_AUTH: API_PASSWORD,
+}
+
+hass = None
+
+
+# pylint: disable=invalid-name
+def setUpModule():
+    """Initialize a Home Assistant server."""
+    global hass
+
+    hass = get_test_home_assistant()
+
+    assert bootstrap.setup_component(
```

Comment:
You don't have to spin up the HTTP server to test your web view. See the generic camera tests https://github.com/home-assistant/home-assistant/blob/dev/tests/components/camera/test_generic.py#L26-L30